code/modules/clothes_making/weaving.dm

```diff
@@ -0,0 +1,383 @@
+	else
+		use_power = MACHINE_POWER_USE_IDLE
+
+/obj/machinery/electric_loom/proc/process_cloth(var/mob/_spinner)
+	if (current_production <= 0)
+		remaining_cloth_to_spin--
+		current_production += matterbin_rating
+
+	current_production--
+	stored_cloth++
+
+	if (stored_cloth >= MAX_SHEET_STACK_AMOUNT)
+		stored_cloth = 0
+		drop_stack(/obj/item/stack/sheet/cloth, get_output(), MAX_SHEET_STACK_AMOUNT)
+		update_icon()
+
+	if (remaining_cloth_to_spin + current_production <= 0)
+		update_icon()
+		return 0
+	return 1
+
+/obj/machinery/electric_loom/update_icon()
+	overlays.len = 0
+	luminosity = 0
+	if (stat & (BROKEN))
+		icon_state = "electric_loom-broken"
+		return
+	if (remaining_cloth_to_spin + current_production <= 0)
+		icon_state = "electric_loom"
+		if (!(stat & (NOPOWER|FORCEDISABLE)) && powered())
+			luminosity = 2
+			var/image/led = image(icon,src,"electric_loom-lightempty")
+			led.plane = ABOVE_LIGHTING_PLANE
+			led.layer = ABOVE_LIGHTING_LAYER
+			overlays += led
+		if (stored_cloth > 0)
+			overlays += "electric_loom-cloth"
+	else if ((stat & (NOPOWER|FORCEDISABLE)) || !powered())
+		icon_state = "electric_loom_ready"
+	else
+		icon_state = "electric_loom_spin"
+		luminosity = 2
+		var/image/led = image(icon,src,"electric_loom-lightready")
+		led.plane = ABOVE_LIGHTING_PLANE
+		led.layer = ABOVE_LIGHTING_LAYER
+		overlays += led
+
+/obj/machinery/electric_loom/power_change()
+	..()
+	update_icon()
+
+/obj/machinery/electric_loom/proc/breakdown()
+	stat |= BROKEN
+	for (var/i = 1 to round(remaining_cloth_to_spin/CLOTH_PER_FLAX))
+		new /obj/item/weapon/reagent_containers/food/snacks/grown/flax(loc)
+	remaining_cloth_to_spin = 0
+	if (stored_cloth > 0)
+		drop_stack(/obj/item/stack/sheet/cloth, loc, stored_cloth)
+		stored_cloth = 0
+		update_icon()
+	update_icon()
+
+/obj/machinery/electric_loom/ex_act(var/severity)
+	switch(severity)
+		if(1)
+			qdel(src)
+		if(2)
+			if (prob(20))
+				qdel(src)
+			else
+				breakdown()
+		if(3)
+			if(prob(50))
+				breakdown()
+
+/obj/machinery/electric_loom/attack_construct(var/mob/user)
+	if(stat & (BROKEN))
+		return
+	if (!Adjacent(user))
+		return 0
+	if(istype(user,/mob/living/simple_animal/construct/armoured))
+		shake(1, 3)
+		playsound(src, 'sound/weapons/heavysmash.ogg', 75, 1)
+		add_hiddenprint(user)
+		breakdown()
+		return 1
+	return 0
+
+/obj/machinery/electric_loom/kick_act(var/mob/living/carbon/human/user)
+	..()
+	if(stat & (BROKEN))
+		return
+	if (prob(5))
+		breakdown()
+
+/obj/machinery/electric_loom/attack_paw(var/mob/user)
+	if(istype(user,/mob/living/carbon/alien/humanoid))
+		if(stat & (BROKEN))
+			return
+		breakdown()
+		user.do_attack_animation(src, user)
+		visible_message("<span class='warning'>\The [user] slashes at \the [src]!</span>")
+		playsound(src, 'sound/weapons/slash.ogg', 100, 1)
+		add_hiddenprint(user)
+	else if (!usr.dexterity_check())
```

Comment:
this is not repeated inside the process, so if your methylin runs out you can continue weaving

also, user not usr

Reply:
weaving on the electric loom isn't a continuous action inlike with the spinning wheel, attack_hand only serves to pick up the current cloth roll so a single dexterity check here is enough.